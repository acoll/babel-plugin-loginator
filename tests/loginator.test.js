const fs = require("fs");
const path = require("path");
const babel = require("babel-core");
const loginator = require("../");

const files = fs.readdirSync(path.join(__dirname, "test-files"));

const babelOpts = {
  plugins: [loginator]
};

describe("File Tests", () => {
  files.forEach(file => {
    test(file, () => {
      const { code } = babel.transformFileSync(
        path.join(__dirname, "test-files", file),
        babelOpts
      );
      expect(code).toMatchSnapshot();
    });
  });
});
