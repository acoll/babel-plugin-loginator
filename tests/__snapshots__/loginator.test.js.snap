// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Tests es-imports.js 1`] = `
"import fs from \\"fs\\";

export default function Thing() {
  console.log(\\"[ LOG ]\\", new Date().toISOString(), \\"[/tests/test-files/es-imports.js:4]\\", \\"Its a thing\\", this);

  return \\"hi\\";
}"
`;

exports[`File Tests function.js 1`] = `
"function sayHi() {
  const x = \\"thing\\";
  console.log(\\"[ LOG ]\\", new Date().toISOString(), \\"[/tests/test-files/function.js:3]\\", \\"hello\\", x);
  return x;
}

console.log(\\"[ LOG ]\\", new Date().toISOString(), \\"[/tests/test-files/function.js:7]\\", \\"function inside console\\", sayHi, sayHi(), () => sayHi());"
`;

exports[`File Tests simple.js 1`] = `"console.log(\\"[ LOG ]\\", new Date().toISOString(), \\"[/tests/test-files/simple.js:1]\\", \\"hello\\");"`;
